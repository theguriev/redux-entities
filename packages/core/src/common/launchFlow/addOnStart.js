export default ({
    dispatch,
    actions: {
        pending
    },
    setState,
    hash,
    key,
    options,
    onStart
}) => ({
    onStart: () => {
        const action = pending({
            hash,
            key
        })
        dispatch(action)
        setState(state => ({
            ...state,
            error: null,
            isLoading: true,
            isRefetching: !!options?.isRefetch
        }))
        onStart(action)
    }
})
