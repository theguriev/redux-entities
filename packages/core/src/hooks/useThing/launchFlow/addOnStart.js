import { Types } from '../utils'

export default ({
    dispatch,
    toType,
    setState,
    hash,
    key,
    options,
    onStart
}) => ({
    onStart: () => {
        const action = {
            type: toType(Types.Pending),
            hash,
            key
        }
        dispatch(action)
        setState(state => ({
            ...state,
            isLoading: true,
            isRefetching: !!options?.isRefetch
        }))
        onStart(action)
    }
})
