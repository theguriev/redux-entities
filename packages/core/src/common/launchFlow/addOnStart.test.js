import {
    launchFlowActions as actions,
    dispatch,
    setState
} from '@redux-things/mocks'
import addOnStart from './addOnStart'

describe('addOnStart', () => {
    test('add onStart option', async () => {
        dispatch.clear()
        let onStartHasBeenCalled = false
        const { onStart } = addOnStart({
            actions,
            hash: 'hash',
            key: 'TKey',
            dispatch,
            setState,
            onStart: result => {
                onStartHasBeenCalled = result
            }
        })
        onStart({ testing: true })
        expect(dispatch.actions).toMatchSnapshot()
        expect(setState.state).toMatchSnapshot()
        expect(onStartHasBeenCalled).toMatchSnapshot()
    })
})
