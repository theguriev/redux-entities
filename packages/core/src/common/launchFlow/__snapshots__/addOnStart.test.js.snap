// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`addOnStart add onStart option 1`] = `
Array [
  Object {
    "hash": "hash",
    "key": "TKey",
    "type": "pending",
  },
]
`;

exports[`addOnStart add onStart option 2`] = `
Object {
  "error": null,
  "initial": true,
  "isLoading": true,
  "isRefetching": false,
}
`;

exports[`addOnStart add onStart option 3`] = `
Object {
  "hash": "hash",
  "key": "TKey",
  "type": "pending",
}
`;
